admindb/sqlite/invites.go

```diff
@@ -0,0 +1,207 @@
+package sqlite
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	"database/sql"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/friendsofgo/errors"
+	"github.com/mattn/go-sqlite3"
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+
+	"github.com/ssb-ngi-pointer/go-ssb-room/admindb"
+	"github.com/ssb-ngi-pointer/go-ssb-room/admindb/sqlite/models"
+	refs "go.mindeco.de/ssb-refs"
+)
+
+// make sure to implement interfaces correctly
+var _ admindb.InviteService = (*Invites)(nil)
+
+// Invites implements the admindb.InviteService.
+// Tokens are stored hashed with sha256 on disk to protect against someone who gaining read-access on the database.
+type Invites struct {
+	db *sql.DB
+
+	allowList *AllowList
+}
+
+const tokenLength = 50
+
+// Create creates a new invite for a new member. It returns the token or an error.
+// createdBy is user ID of the admin or moderator who created it.
+// aliasSuggestion is optional (empty string is fine) but can be used to disambiguate open invites. (See https://github.com/ssb-ngi-pointer/rooms2/issues/21)
+// The returned token is base64 URL encoded and has tokenLength when decoded.
+func (i Invites) Create(ctx context.Context, createdBy int64, aliasSuggestion string) (string, error) {
+	isValidSuggestion := aliasSuggestion != ""
+	var newInvite = models.Invite{
+		CreatedBy:       createdBy,
+		AliasSuggestion: null.NewString(aliasSuggestion, isValidSuggestion),
```

Comment:
`null.NewString` *does a double take*, [oh](https://github.com/volatiletech/null)

Reply:
We could possible just store empty strings. They don't _need_ to be stored as NULLs when not present.